src/sdl2_compat.c

```diff
@@ -681,18 +690,80 @@ SDL_GetHintBoolean(const char *name, SDL2_bool default_value)
     return SDL3_GetHintBoolean(SDL2_to_SDL3_hint(name), default_value) ? SDL2_TRUE : SDL2_FALSE;
 }
 
-/* FIXME: callbacks may need tweaking ... */
+static void SDLCALL hint_callback_wrapper(void *userdata, const char *name, const char *oldValue, const char *newValue)
+{
+    HintCallbackInfo *info = (HintCallbackInfo *) userdata;
+    info->callback(info->userdata, info->name, oldValue, newValue);
```

Comment:
We need to convert `oldValue` and `newValue` with `SDL2_to_SDL3_hint_value()` too.

Reply:
Can you please re-review my latest iteration?
old and new names are converted, goes all-in on properties and no longer uses linked lists.